src/cljs/athens/types/tasks/view.cljs

```diff
@@ -0,0 +1,336 @@
+
+         (if invalid-prop-str?
+           [:> FormErrorMessage
+            (str prop-title " is " (if required?
+                                     "required"
+                                     "empty"))]
+           [:> FormHelperText
+            (str "Please provide " prop-title)])]))))
+
+
+(defn- find-allowed-statuses
+  []
+  (let [task-status-page    (reactive/get-reactive-node-document [:node/title ":task/status"])
+        allowed-stat-blocks (-> task-status-page
+                                :block/properties
+                                (get ":property/enum")
+                                :block/children)
+        allowed-statuses    (map #(select-keys % [:block/uid :block/string]) allowed-stat-blocks)]
+    (if (seq allowed-statuses)
+      allowed-statuses
+      (let [[statuses ops] (create-default-allowed-statuses @db/dsdb)
+            event          (common-events/build-atomic-event ops)]
+        (rf/dispatch [:resolve-transact-forward event])
+        statuses))))
+
+
+(defn task-status-view
+  [parent-block-uid status-block-uid]
+  (let [status-id        (str (random-uuid))
+        status-block     (reactive/get-reactive-block-document [:block/uid status-block-uid])
+        allowed-statuses (find-allowed-statuses)
+        status-string    (:block/string status-block "(())")
+        status-uid       (subs  status-string 2 (- (count status-string) 2))]
+    [:> FormControl {:is-required true}
+     [:> HStack {:spacing "2rem"}
+      [:> FormLabel {:html-for status-id
+                     :w        "9rem"}
+       "Task Status"]
+      [:> Select {:id          status-id
+                  :value       status-uid
+                  :placeholder "Select a status"
+                  :on-change   (fn [e]
+                                 (let [new-status (-> e .-target .-value)]
+                                   (rf/dispatch [::task-events/save-status
+                                                 {:parent-block-uid parent-block-uid
+                                                  :status           (str "((" new-status "))")}])))}
+       (doall
+         (for [{:block/keys [uid string]} allowed-statuses]
+           ^{:key uid}
+           [:option {:value uid}
+            string]))]]]))
+
+
+(defn- find-property-block-by-key-name
+  [entity-block prop-name]
+  (->> entity-block
+       :block/properties
+       (filter (fn [[k _v]] (= prop-name k)))
+       (map second)
+       first))
+
+
+(defrecord TaskView
+  []
+
+  types/BlockTypeProtocol
+
+  (inline-ref-view
+    [_this _block-data _attr _ref-uid _uid _callbacks _with-breadcrumb?])
+
+
+  (outline-view
+    [_this block-data _callbacks]
+    (let [block-uid (:block/uid block-data)]
+      (fn [_this _block-data _callbacks]
+        (let [reactive-block  (reactive/get-reactive-block-document [:block/uid block-uid])
+              title-uid       (:block/uid (find-property-block-by-key-name reactive-block ":task/title"))
```

Comment:
I should take a look at this, it should be easier, ideally just destructure directly.